Lombiq.Tests.UI/Extensions/VisualVerificationUITestContextExtensions.cs

```diff
@@ -0,0 +1,509 @@
+    /// <param name="roi">Region of interest. Can be  null.</param>
+    /// <param name="configurator">Action callback to configure the behavior. Can be null.</param>
+    /// <exception cref="VisualVerificationReferenceImageCreatedException">
+    /// If no reference image found under project path.
+    /// </exception>
+    /// <exception cref="VisualVerificationAttributeNotFoundException">
+    /// If no method found annotated with <see cref="VisualVerificationAttribute"/> in stacktrace.
+    /// </exception>
+    public static void VisualAssertApproved(
+        this UITestContext context,
+        IWebElement element,
+        double meanErrorPercentageThreshold,
+        Rectangle? roi = null,
+        Action<VisualMatchApprovedConfiguration> configurator = null) => context
+        .VisualAssertApproved(
+            element,
+            (approvedContext, diff) =>
+                AssertInternal(
+                    approvedContext,
+                    (actual, expected) => actual <= expected,
+                    diff.PixelErrorPercentage,
+                    meanErrorPercentageThreshold,
+                    nameof(diff.PixelErrorPercentage),
+                    ConditionLessThenOrEqual),
+            roi,
+            configurator);
+
+    /// <summary>
+    /// Compares the reference image and screenshot of the element given by <paramref name="elementSelector"/>.
+    /// The reference image is automatically loaded from assembly resource, if it's not exists then from the project
+    /// path based on <see cref="VisualMatchApprovedConfiguration"/> - it can be configured over
+    /// <paramref name="configurator"/> -, if the reference image doesn't exist, a new one will be created based on the
+    /// element's screenshot, and an <see cref="VisualVerificationReferenceImageCreatedException"/> will be thrown. The
+    /// reference image path is generated from the method name - annotated with
+    /// <see cref="VisualVerificationAttribute"/> - and the source file name and path, where the method is.
+    /// </summary>
+    /// <param name="context">The <see cref="UITestContext"/> in which the extension is executed on.</param>
+    /// <param name="elementSelector">Selector for the target element.</param>
+    /// <param name="comparator">To validate the comparison result.</param>
+    /// <param name="roi">Region of interest. Can be  null.</param>
+    /// <param name="configurator">Action callback to configure the behavior. Can be null.</param>
+    /// <exception cref="VisualVerificationReferenceImageCreatedException">
+    /// If no reference image found under project path.
+    /// </exception>
+    /// <exception cref="VisualVerificationAttributeNotFoundException">
+    /// If no method found annotated with <see cref="VisualVerificationAttribute"/> in stacktrace.
+    /// </exception>
+    private static void VisualAssertApproved(
```

Comment:
Private methods' API doesn't need to be documented to such a degree. Or did you intend this to be public? Also see the method below.

Reply:
This was originally `public`, but at the end, I changed it to `private ` because of the `comparator` callback. It throws the exception finally. Now, I would keep it in this scope.

Reply:
Why would you keep the documentation when it's not needed? It's something we'd need to keep maintaining.